Source/WebCore/platform/mediarecorder/cocoa/MediaRecorderPrivateWriterWebM.mm

```diff
@@ -262,13 +262,21 @@ static String codecStringForMediaVideoCodecId(FourCharCode codec)
 }
 
 void MediaRecorderPrivateWriterWebM::appendVideoFrame(VideoFrame& frame)
+{
+    ensureOnMainThread([weakThis = ThreadSafeWeakPtr { *this }, frame = Ref { frame }]() mutable {
```

Comment:
We had issues in the past with main thread being blocked, which would exhaust the video frame pool.
Situation might be better now, please have a look though at what is being done in MediaPlayerPrivateMediaStreamAVFObjC::processNewVideoFrame to only keep one strong video frame reference at a time.

Reply:
I hadn't considered that the main thread could be too busy to process new frames. This would result in an incomplete recording ; something the mp4 recorder wouldn't suffer from as it's all done asynchronously in the GPU process.

I'll move the writer to use it's own task queue then.

Reply:
This change was part of an earlier commit that I included for ease of testing/rebase